lib/rules/no-prefix-styled-component.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * @fileoverview Prevent the declaration of a styled component prop without the dollar sign prefix
+ * @author Max Balej
+ */
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    type: "problem",
+    docs: {
+      description:
+        "Prevent the declaration of a styled component prop without the dollar sign prefix",
+      category: "Wrong component usage",
+      recommended: true,
+    },
+    messages: {
+      prefixError: "Expected prefix '$' for styled-components props.",
+    },
+    fixable: null,
+    schema: [],
+  },
+
+  create: function (context) {
+    const styleIdentifier = "styled";
+
+    function unwrapExpression(node) {
+      const callExpression = node?.tag?.callee?.name === styleIdentifier;
+      const memberExpression = node?.tag?.object?.name === styleIdentifier;
```

Comment:
peut-être un commentaire pour expliciter, wdyt ? Exemple:
```suggestion
      const callExpression = node?.tag?.callee?.name === styleIdentifier; // styled(<comp>)
      const memberExpression = node?.tag?.object?.name === styleIdentifier; // styled.<tag>
```